lib/facter/models/fact_collection.rb

```diff
@@ -17,13 +23,18 @@ def build_fact_collection!(facts)
       self
     end
 
+    def dig_fact(user_query)
+      split_user_query = Facter::Utils.split_user_query(user_query)
+      dig(*split_user_query) || dig(user_query)
+    rescue NoMethodError, TypeError
+      # An incorrect user query (e.g. mountpoints./.available.asd) can cause
+      # Facter to call dig on a string, which raises a type error.
+      # If this happens, we assume the query is wrong and silently continue.
+      nil
```

Comment:
if this is not happening often, we could log a debug message with the exception message instead of silently ignoring exception

Reply:
updated to always log if dig result is nil, or if we get an exception